src/integrationTest/samVisualize/visualizeSamTemplate.test.ts

```diff
@@ -0,0 +1,152 @@
+        assert.deepStrictEqual(result.viewType, 'samVisualization')
+        assert.ok(result.webview.html)
+    })
+
+    it('update webview when the editor is changed', async function () {
+        const textEditor = await openATextEditorWithText('Sample Text', sampleSamTemplateFileName)
+
+        const result = await vscode.commands.executeCommand<vscode.WebviewPanel>('aws.samVisualize.renderTemplate')
+
+        assert.ok(result)
+
+        const didViewStateChange = new Promise<vscode.WebviewPanelOnDidChangeViewStateEvent>(resolve => {
+            result.onDidChangeViewState(event => {
+                assert.ok(event)
+                resolve(event)
+            })
+        })
+
+        // Edit the editor to replace the entire file with a working SAM template
+        await textEditor.edit(edit => {
+            edit.replace(
+                new vscode.Range(
+                    textEditor.document.positionAt(0),
+                    textEditor.document.positionAt(textEditor.document.getText().length)
+                ),
+                sampleSamTemplateYaml
+            )
+        })
+
+        await didViewStateChange
+    })
+
+    it('throws an error if no text editor is open', async function () {
+        // Ensure no text editor is open
+        await closeAllEditors()
+
+        await assert.rejects(async () => {
+            await vscode.commands.executeCommand<vscode.WebviewPanel>('aws.samVisualize.renderTemplate')
+        })
+    })
+
+    it('doesnt update the graph if a seperate file is opened or modified', async function () {
+        await openATextEditorWithText(sampleSamTemplateYaml, sampleSamTemplateFileName)
+
+        const result = await vscode.commands.executeCommand<vscode.WebviewPanel>('aws.samVisualize.renderTemplate')
+        assert.ok(result)
+
+        // Here we will create a second file unrelated to the existing graph in VS Code.
+
+        const someOtherFileName = 'other.yaml'
+        const someOtherFileText = 'Random contents'
+
+        const textEditor2 = await openATextEditorWithText(someOtherFileText, someOtherFileName)
+
+        const updatedText = 'updated text'
+
+        const postMessageSpy = spy()
+        if (result) {
+            // eslint-disable-next-line @typescript-eslint/unbound-method
+            result.webview.postMessage = postMessageSpy
+        }
+
+        // Update and save the file that we are NOT visualizing
+        await textEditor2.edit(eb => {
+            eb.replace(
+                new vscode.Range(
+                    textEditor2.document.positionAt(0),
+                    textEditor2.document.positionAt(textEditor2.document.getText().length)
+                ),
+                updatedText
+            )
+        })
+
+        // Save the file
+        await textEditor2.document.save()
+
+        // The visualization should not update.
+        assert(postMessageSpy.notCalled)
```

Comment:
using Spy in integration tests seems a bit contradictory.

Reply:
This test is directly analogous to a test in `/src/integrationTest/stepFunctions/visualizeStateMachine.test.ts`, where the exact same functionality is tested on the stepFunction visualization. Is there a better way ensure postMessages are not sent?